### PR TITLE
added new react templates, github CI and terraform generators

### PR DESCRIPTION
**Pull request by zthomas**

### CODE REVIEW

File: templates/ci/github.tmpl [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    - run: npm test
+    
+    - run: npm run build
+
+    - name: upload build artifacts
+      uses: actions/upload-artifact@v1
+      with:
+        name: build
+        path: build
```
Comment (bmonkman) [inline templates/ci/github.tmpl]: Not important now but you should make sure you have "add newline at end of file" turned in in your editor.

File: templates/react/.github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Download build artifacts
+        uses: actions/download-artifact@v1
+        with:
+          name: build
+
+      # https://github.com/opspresso/action-s3-sync
+      - name: Publish to AWS S3
+        uses: opspresso/action-s3-sync@master
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (ShahNewazKhan) [inline templates/react/.github/workflows/deploy.yml]: where is the secrets struct being hydrated?

**Comment by bmonkman** [comment]
Feel free to just

File: templates/react/.github/workflows/deploy.yml (hunk 2 above)
Comment (zthomas) [reply templates/react/.github/workflows/deploy.yml]: Github Secrets. You just have to add them through their UI.

Comment (ShahNewazKhan) [reply templates/react/.github/workflows/deploy.yml]: Right that's why I couldn't find it anywhere  :+1:

**Pull request merged**